### ISSUE TITLE
Which token ID is used to represent the empty token in the delay pattern?

### ISSUE THREAD

**Issue by yangdongchao** [kyutai-labs/moshi#123]
### Due diligence

- [X] I have done my due diligence in trying to find the answer myself.

### Topic

The paper

### Question

Hi, I want to ask which token ID is used to represent the empty token in the delay pattern? In your Figure 4, I see you use token 0 denotes the empty token. I want to confirm that whether 0 is used as the empty token during training? This token will be considered when calculating the loss?

**Comment by LaurentMazare** [comment]
We use token 0 (EPAD in the paper) and 3 (PAD) as in figure 4 page 15. The weight of padding tokens are reduced in the training loss, see "Moshi pre-training" in section 4.4 page 20.

**Comment by yangdongchao** [comment]
> We use token 0 (EPAD in the paper) and 3 (PAD) as in figure 4 page 15. The weight of padding tokens are reduced in the training loss, see "Moshi pre-training" in section 4.4 page 20.

Thank you your reply. 
I can understand that you use EPAD and PAD to padding the text sequence. My concern about the empty token in the acoustic token.  For example, taking timesteps=4 and n_q=3, delays=1, the multi-codebook sequence:
        [[text_1,          text_2,          text_3,          text_4],
        [Semantic_1,  Semantic_2, Semantic_3, Semantic_4],
        [Acoustic_1,   Acoustic_2,  Acoustic_3,   Acoustic_4]]
        The resulting sequence obtained from the returned pattern is:
        [[ text_1,  text_2,         text_3,          text_4,           S],
        [ S,          Semantic_1,  Semantic_2, Semantic_3,  Semantic_4],
        [ S,          Acoustic_1,    Acoustic_2,  Acoustic_3,   Acoustic_4]]

So, my question is, which token will be used as S. Because I want to finetuning your checkpoint, so I want to keep the same token with you.

**Comment by LaurentMazare** [comment]
Token 2048 is used for the initial timesteps of the audio codebooks.

**Comment by yangdongchao** [comment]
> Token 2048 is used for the initial timesteps of the audio codebooks.

Thank you for your reply.
